main/lsp/LSPTypechecker.cc

```diff
@@ -503,7 +503,13 @@ LSPTypechecker::SlowPathResult LSPTypechecker::runSlowPath(LSPFileUpdates &updat
                     // make a copy to pass back over.
                     ENFORCE(this->gs->packageDB().packages().empty(),
                             "Don't want symbols or packages in indexer GlobalState");
-                    indexedState = this->gs->deepCopyGlobalState();
+                    indexedState = this->gs->copyForIndex(
```

Comment:
The indexer thread only really requires a copy of the file table, and `copyForIndex` is the most convenient way to get that currently. I also like the symmetry with `pipeline::index`.